src/ui/actions/network.ts

```diff
@@ -126,3 +129,31 @@ export function selectAndFetchRequest(requestId: RequestId): UIThunkAction {
     });
   };
 }
+
+export function seekToRequestFrame(
+  request: RequestSummary,
+  frameId: string,
+  cx: Context
+): UIThunkAction {
+  return async (dispatch, getState, { ThreadFront }) => {
+    const state = getState();
+    const loadedRegions = getLoadedRegions(state);
+    const point = request.point;
+
+    // Don't select a request that's not within a loaded region.
+    if (!request || !loadedRegions || !isPointInRegions(loadedRegions.loaded, point.point)) {
+      return;
+    }
+
+    ThreadFront.timeWarp(point.point, point.time, true);
+    const frames = (await ThreadFront.getFrames()) || [];
+    const formattedFrames = await Promise.all(frames?.map((frame, i) => createFrame(frame, i)));
+    const frame = formattedFrames.find(f => f?.id === frameId);
+    if (!frame) {
+      console.error(`Request frame not found: ${frameId}`);
+      return;
+    }
+    await (new Promise<void>((resolve) => setTimeout(resolve, 50)));
```

Comment:
@hbenl help! I'm having timing issues where because I create a pause the editor changes location, but then I immediately select a source, and for some reason they are getting processed in the wrong order. Doing this solves it but... it feels pretty gross. Is there any way to say "Pause here" without also saying "and go to the editor location of the pause"?

Reply:
sorry @souporserious I want to get this one thing taken care of and then I'll get this out!

Reply:
No worries at all! Just using the preview branch to get the stack trace for storyboarding 😎 thank you again for tackling this so quick!

Reply:
OK, I think I took care of this, I now allow `timeWarp` to take a frame to go to, rather than automatically going to the top frame of the acquired pause (but we will still go there if no frame is provided).

Reply:
That sounds like the right solution to me.